tests/bug904.js

```diff
@@ -0,0 +1,6 @@
+import {assert, assertThrows} from "./assert.js"
+let calls = 0
+Error.prepareStackTrace = function() { calls++ }
+function f() { f() }
+assertThrows(RangeError, f)
+assert(calls, 0)
```

Comment:
The fact the number of calls is zero instead of one is currently more or less by accident but in the case of stack overflow that's the desired behavior IMO (no JS calls) and I have an upcoming pull request that makes that intentional.